machine.py

```diff
@@ -0,0 +1,121 @@
+from typing import Dict, Set
+
+from coin import Coin
+from costumer import Costumer
+from drink import Drink
+from resource import Resource
+
+
+def print_coins(coins: Dict[Coin, int]):
+    for coin in coins.keys():
+        print(f'{coin.name}: {coins[coin]}')
+
+
+class NoSuchDrinkError(Exception):
```

Comment:
It's quite common to create MachineError(Exception) base exception and  make all exceptions originating from machine module inherit it, such that external code could handle general machine error.